packages/core/src/browser/saveable.ts

```diff
@@ -50,24 +52,31 @@ export namespace Saveable {
     export async function save(arg: any): Promise<void> {
         const saveable = getDirty(arg);
         if (saveable) {
+            saveable.saving = true;
```

Comment:
You cannot assume that someone else won't call `save`. In this case, `saving` flag will be ignored. `Saveable`should manage it.